### PR TITLE
Feature/#42 update delete node test

### PR DESCRIPTION
**Pull request by pbimpisidis**
DeleteNode.Tests should be fixed according to the comments

can you check only the DeleteNode.Test? I don't know why I got the other files showing up here too..

### CODE REVIEW

**Comment by rufer7** [comment]
@pbimpisidis I don't get it. In pull request #39 there is already a file `DeleteNode.Tests.ps1`...

**Comment by pbimpisidis** [comment]
@rufer7  ok I removed DeleteNode.Tests.ps1 from #37

File: it/DeleteNode.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            }
+        }
+		
+		It "DeleteNodeWithChildNode" -Test {
+			#ARRANGE
+			$nodeName = $entityPrefix + "node";
+			$childName = $entityPrefix + "childnode";
+			
+			#ACT create node
+			$newNode = New-ApcNode -Name $nodeName -ParentId 1 -EntityKindId 1 -svc $svc | select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis use constants from public module here as well

```
$nodeEntityKindId = [biz.dfch.CS.Appclusive.Public.Constants+EntityKindId]::Node.value__;
$nodeParentId = (Get-ApcTenant -Current).NodeId;
```

File: it/DeleteNode.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+			
+			#get Id of the node
+			$nodeId = $newNode.Id;
+			
+			#ACT create child Node
+			$newChildNode = New-ApcNode -Name $childName -ParentId $nodeId -EntityKindId 1 -svc $svc | select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis use constants from public module here as well

```
$nodeEntityKindId = [biz.dfch.CS.Appclusive.Public.Constants+EntityKindId]::Node.value__;
$nodeParentId = (Get-ApcTenant -Current).NodeId;
```

File: it/DeleteNode.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+		It "DeleteNodeCheckAttatchedEntitiesDeletion" -Test {
+			#ARRANGE
+			$nodeName = $entityPrefix + "node";
+			$extName = $entityPrefix + "externalnode";
+			$aclName = $entityPrefix + "Acl";
+			$aceName = $entityPrefix + "Ace";
+			$entityBagName = $entityPrefix + "entityBag";
+			
+			#ACT create node
+			$newNode = New-ApcNode -Name $nodeName -ParentId 1 -EntityKindId 1 -svc $svc| select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis use constants from public module here as well

```
$nodeEntityKindId = [biz.dfch.CS.Appclusive.Public.Constants+EntityKindId]::Node.value__;
$nodeParentId = (Get-ApcTenant -Current).NodeId;
```

File: it/DeleteNode.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+			$result = $svc.Core.SaveChanges();
+			
+			#ASSERT
+			#check that node is deleted
+			$node = Get-ApcNode -Id $nodeId -svc $svc | select;
+			$node | Should Be $null;
+			
+			#check that Job is deleted
+			$query = "Id eq {0}" -f $jobId;
+			$job = $svc.Core.Jobs.AddQueryOption('$filter', $query) | select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis fyi you could use Get-ApcJob instead

File: it/DeleteNode.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+			$result = $svc.Core.SaveChanges();
+			
+			#get EntityBag
+			$query = "Name eq '{0}' and EntityId eq {1}" -f $entityBagName, $nodeId;
+			$entityBag = $svc.Core.EntityBags.AddQueryOption('$filter', $query) | select;
+			$entityBag | Should Not Be $null;
+			$entityBagId = $entityBag.Id;
+			
+			#ACT delete Node
+			$node = Get-ApcNode -Id $nodeId -svc $svc | select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

File: it/Acl_Ace.Tests.ps1 [hunk 6]
```diff
@@ -8,385 +11,321 @@ function Stop-Pester($message = "EMERGENCY: Script cannot continue.")
...
+        }
+		
+		It "Acl-CreateAndDelete" -Test {
+		
+			#ARRANGE
+			$nodeName = $entityPrefix + "newtestnode";
+			$aclName = $entityPrefix + "Acl";
+			
+			#ACT create node
+			$newNode = New-ApcNode -Name $nodeName -ParentId 1 -EntityKindId 1 | select;
```
Comment (rufer7) [inline it/Acl_Ace.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here as the node gets returned automatically

File: it/Acl_Ace.Tests.ps1 [hunk 7]
```diff
@@ -8,385 +11,321 @@ function Stop-Pester($message = "EMERGENCY: Script cannot continue.")
...
+			
+			#get Id and entityKindId of the node
+			$nodeId = $newNode.Id;
+			$nodeEntityKindId = $newNode.EntityKindId;
+			
+			#ACT create acl
+			$acl = Create-Acl -svc $svc -aclName $aclName -entityId $nodeId -entityKindId $nodeEntityKindId | select;
```
Comment (rufer7) [inline it/Acl_Ace.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

File: it/Acl_Ace.Tests.ps1 [hunk 8]
```diff
@@ -8,385 +11,321 @@ function Stop-Pester($message = "EMERGENCY: Script cannot continue.")
...
+			
+			#ACT create node
+			$newNode = New-ApcNode -Name $nodeName -ParentId 1 -EntityKindId 1 | select;
+			
+			#get Id and EntityKindId of the node
+			$nodeId = $newNode.Id;
+			$nodeEntityKindId = $newNode.EntityKindId;
+			
+			#ACT create acl
+			$acl = Create-Acl -svc $svc -aclName $aclName -entityId $nodeId -entityKindId $nodeEntityKindId | select;
```
Comment (rufer7) [inline it/Acl_Ace.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

File: it/DeleteNode.Tests.ps1 (hunk 1 above)
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

File: it/DeleteNode.Tests.ps1 (hunk 2 above)
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

File: it/DeleteNode.Tests.ps1 [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+			$nodeId = $newNode.Id;
+			
+			#ACT create child Node
+			$newChildNode = New-ApcNode -Name $childName -ParentId $nodeId -EntityKindId 1 -svc $svc | select;
+			
+			#get Id of the child Node
+			$childNodeId = $newChildNode.Id;
+			
+			#get parent & child Node
+			$parentNode = Get-ApcNode -Id $nodeId -svc $svc | select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

File: it/DeleteNode.Tests.ps1 [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+			$childNode = Get-ApcNode -Id $childNodeId -svc $svc | select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

File: it/DeleteNode.Tests.ps1 [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+			$parentNode | Should Not Be $null;
+			$parentNode.Id | Should Not Be $null;
+			$childNode | Should Not Be $null;
+			$childNode.Id | Should Not Be $null;
+			$childNode.ParentId | Should Be $nodeId;
+			
+			try
+			{
+				#get the parent node
+				$parentNode = Get-ApcNode -Id $nodeId -svc $svc | select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

File: it/DeleteNode.Tests.ps1 [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+				$(Format-ApcException) | Should Not Be $null;
+				$_.Exception.Message | Should Not Be $null;
+				$_.FullyQualifiedErrorId | Should Not Be $null;
+			}
+			finally 
+			{
+				$svc = Enter-Appclusive;
+				
+				#get the child node
+				$childNode = Get-ApcNode -Id $childNodeId -svc $svc | select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

File: it/DeleteNode.Tests.ps1 [hunk 13]
```diff
@@ -0,0 +1,196 @@
...
+	
+				#delete the child node first
+				$svc.Core.DeleteObject($childNode);
+				$result = $svc.Core.SaveChanges();
+				
+				#get the parent node
+				$parentNode = Get-ApcNode -Id $nodeId -svc $svc | select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

File: it/DeleteNode.Tests.ps1 [hunk 14]
```diff
@@ -0,0 +1,196 @@
...
+			
+			#ACT create node
+			$newNode = New-ApcNode -Name $nodeName -ParentId 1 -EntityKindId 1 -svc $svc| select;
+			
+			#get Id and entityKindId of the node
+			$nodeId = $newNode.Id;
+			$nodeEntityKindId = $newNode.EntityKindId;
+			
+			#get the job of the node
+			$job = Get-ApcNode -Id $nodeId -ExpandJob | select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

File: it/DeleteNode.Tests.ps1 [hunk 15]
```diff
@@ -0,0 +1,196 @@
...
+			$extNode = New-ApcExternalNode -name $extName -NodeId $nodeId -ExternalId $nodeId -ExternalType "ArbitraryType" -svc $svc | select;
+			#Write-Host ($extNode2 | out-string);
+			
+			#ASSERT external Node
+			$extNode | Should Not Be $null;
+			#get id of external node
+			$extNodeId = $extNode.Id;
+			
+			#ACT create acl
+			$acl = Create-Acl -svc $svc -aclName $aclName -entityId $nodeId -entityKindId $nodeEntityKindId | select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

File: it/DeleteNode.Tests.ps1 [hunk 16]
```diff
@@ -0,0 +1,196 @@
...
+			
+			#get Id of the acl
+			$aclId = $acl.Id;
+			
+			#ACT Create Ace
+			$ace = Create-Ace -svc $svc -aceName $aceName -aclId $aclId | select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

File: it/DeleteNode.Tests.ps1 [hunk 17]
```diff
@@ -0,0 +1,196 @@
...
+			$entityBagId = $entityBag.Id;
+			
+			#ACT delete Node
+			$node = Get-ApcNode -Id $nodeId -svc $svc | select;
+			$svc.Core.DeleteObject($node);
+			$result = $svc.Core.SaveChanges();
+			
+			#ASSERT
+			#check that node is deleted
+			$node = Get-ApcNode -Id $nodeId -svc $svc | select;
```
Comment (rufer7) [inline it/DeleteNode.Tests.ps1]: @pbimpisidis as far as I know is `| Select` not needed here

Comment (pbimpisidis) [inline it/DeleteNode.Tests.ps1]: removed

File: it/DeleteNode.Tests.ps1 (hunk 16 above)
Comment (pbimpisidis) [inline it/DeleteNode.Tests.ps1]: removed

File: it/DeleteNode.Tests.ps1 (hunk 15 above)
Comment (pbimpisidis) [inline it/DeleteNode.Tests.ps1]: removed

File: it/DeleteNode.Tests.ps1 (hunk 14 above)
Comment (pbimpisidis) [inline it/DeleteNode.Tests.ps1]: removed

File: it/DeleteNode.Tests.ps1 (hunk 13 above)
Comment (pbimpisidis) [inline it/DeleteNode.Tests.ps1]: removed

File: it/DeleteNode.Tests.ps1 (hunk 12 above)
Comment (pbimpisidis) [inline it/DeleteNode.Tests.ps1]: removed

File: it/DeleteNode.Tests.ps1 (hunk 11 above)
Comment (pbimpisidis) [inline it/DeleteNode.Tests.ps1]: removed

File: it/DeleteNode.Tests.ps1 (hunk 10 above)
Comment (pbimpisidis) [inline it/DeleteNode.Tests.ps1]: removed

File: it/DeleteNode.Tests.ps1 (hunk 9 above)
Comment (pbimpisidis) [inline it/DeleteNode.Tests.ps1]: removed

File: it/DeleteNode.Tests.ps1 (hunk 5 above)
Comment (pbimpisidis) [inline it/DeleteNode.Tests.ps1]: removed

File: it/DeleteNode.Tests.ps1 (hunk 2 above)
Comment (pbimpisidis) [inline it/DeleteNode.Tests.ps1]: removed select & used constant

File: it/DeleteNode.Tests.ps1 (hunk 1 above)
Comment (pbimpisidis) [inline it/DeleteNode.Tests.ps1]: removed select & used constant

File: it/DeleteNode.Tests.ps1 (hunk 3 above)
Comment (pbimpisidis) [inline it/DeleteNode.Tests.ps1]: applied

**Pull request merged**